client/network/sync/src/engine.rs

```diff
@@ -896,10 +912,33 @@ where
 				);
 			}
 
-			return Err(())
+			return Err(true)
 		}
 
-		let no_slot_peer = self.default_peers_set_no_slot_peers.contains(&peer);
+		Ok(handshake)
+	}
+
+	/// Validate inbound connection.
```

Comment:
The documentation comment is misleading: this function validates both `inbound` and `outbound` connections.